v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/model/helpers/JobSpecMapper.java

```diff
@@ -0,0 +1,106 @@
+                jobSpecRequest.config = new Config(jobSpecObj.getJSONObject("config"));
+            }
+
+            if (jobSpecObj.has("source")) {
+                Source source = SourceMapper.fromJson(jobSpecObj.getJSONObject("source"));
+                if (StringUtils.isNotEmpty(source.name)) {
+                    jobSpecRequest.sources.put(source.name, source);
+                } else {
+                    jobSpecRequest.sources.put(DEFAULT_SOURCE_NAME, source);
+                }
+            } else if (jobSpecObj.has("sources")) {
+
+                JSONArray sourceArray = jobSpecObj.getJSONArray("sources");
+                for (int i = 0; i < sourceArray.length(); i++) {
+                    final Source source = SourceMapper.fromJson(sourceArray.getJSONObject(i));
+                    if (StringUtils.isNotEmpty(source.name)) {
+                        jobSpecRequest.sources.put(source.name, source);
+                    } else {
+                        jobSpecRequest.sources.put(DEFAULT_SOURCE_NAME, source);
+                    }
+                }
+            } else {
+                // there is no source defined this could be used in a big query job...
+                // this would lead to a validation error (elsewhere)
+            }
+
+            if (jobSpecObj.has("targets")) {
+                final JSONArray targetObjArray = jobSpecObj.getJSONArray("targets");
+                for (int i = 0; i < targetObjArray.length(); i++) {
+                    final Target target = TargetMapper.fromJson(targetObjArray.getJSONObject(i));
+                    jobSpecRequest.targets.add(target);
+                }
+            }
+
+            //Note: this options array was created to allow mimicing Spark syntax.
+            //It is currently unused.
+            if (jobSpecObj.has("options")) {
+                final JSONArray optionsArray = jobSpecObj.getJSONArray("options");
+                for (int i = 0; i < optionsArray.length(); i++) {
+                    JSONObject jsonObject = optionsArray.getJSONObject(i);
+                    Iterator<String> keys = jsonObject.keys();
+                    while (keys.hasNext()) {
+                        String key = keys.next();
+                        jobSpecRequest.options.put(key, jsonObject.getString(key));
+                    }
+                }
+            }
+
+            if (jobSpecObj.has("actions")) {
+                final JSONArray optionsArray = jobSpecObj.getJSONArray("actions");
+                for (int i = 0; i < optionsArray.length(); i++) {
+                    JSONObject jsonObject = optionsArray.getJSONObject(i);
+                    Action action = ActionMapper.fromJson(jsonObject);
+                    jobSpecRequest.actions.add(action);
+                }
+            }
+
+        } catch (final Exception e) {
+            LOG.error(
+                    "Unable to parse beam configuration from {}: ", jobSpecUri, e);
```

Comment:
IIRC, there no LOG.error(String, Object, Exception) method, i.e. the last argument `e` will be treated as an object, which is not referenced in the message pattern (1st argument), and thus will be ignored.